locals.tf

```diff
@@ -31,6 +31,34 @@ locals {
   nfs_vm_subnet    = var.create_nfs_public_ip ? module.vpc.public_subnets[0] : module.vpc.private_subnets[0]
   nfs_vm_subnet_az = var.create_nfs_public_ip ? module.vpc.public_subnet_azs[0] : module.vpc.private_subnet_azs[0]
 
+  # Generate list of AZ where created subnets should be placed
+  # If not specified by the user replace with list of all AZs in a region
+  # If not enough regions provided, append with the list of all AZs in the region while retaining
+  # order of user provided list of regions
+  public_subnet_azs   = (
+  can(var.subnet_azs["public"]) ?
+    (length(var.subnet_azs["public"]) >= length(lookup(var.subnets, "public", [])) ?
+      var.subnet_azs["public"]
+      : distinct(concat(var.subnet_azs["public"], data.aws_availability_zones.available.names)))
+    : data.aws_availability_zones.available.names
+  )
+
+  private_subnet_azs   = (
+  can(var.subnet_azs["private"]) ?
+    (length(var.subnet_azs["private"]) >= length(lookup(var.subnets, "private", [])) ?
+      var.subnet_azs["private"]
+      : distinct(concat(var.subnet_azs["private"], data.aws_availability_zones.available.names)))
+    : data.aws_availability_zones.available.names
+  )
+
+  database_subnet_azs   = (
+  can(var.subnet_azs["database"]) ?
+    (length(var.subnet_azs["database"]) >= length(lookup(var.subnets, "database", [])) ?
+      var.subnet_azs["database"]
+      : distinct(concat(var.subnet_azs["database"], data.aws_availability_zones.available.names)))
+    : data.aws_availability_zones.available.names
+  )
+
```

Comment:
The code now takes what the customer provided an augmetns that value, i.e. its not what the customer provided, its a very well educated guess and provides what's needed, but it simply covers up the user error in that they did not provide the correct number of items for this element. Need a discussion on this.

Reply:
I agree, it may be better not to correct or append to the user input and let it fail, the user should correct any mistakes made.
This should be a quick update to the `local.tf` + doc , I'll make the change and test it out.

Reply:
Updated the behavior, no longer correcting user input and will let it fail instead. The user should be responsible for correct input. Docs and locals.tf changed, performing tests.